### ISSUE TITLE
pack -> buildpack not found

### ISSUE THREAD

**Issue by nateberkopec** [hashicorp/waypoint#1349]
Given the following `waypoint.hcl`:

```
project = "web"

app "web" {
  build {
    use "pack" {
      builder = "heroku/buildpacks:20"
      buildpacks = [
        "urn:cnb:registry:fagiani/apt"
      ]
    }
  }
}
```

... I get:

```
❌ Building image
 │ 2021/04/16 13:05:33.976595 DEBUG:  Pulling image index.docker.io/heroku/buildpac
 │ ks:20
 │ 20: Pulling from heroku/buildpacks
 │ Digest: sha256:b72e666e07185b2271638461f0d5b33d3fb415dd93c9c55037957da6db09ed85
 │ Status: Image is up to date for heroku/buildpacks:20
 │ 2021/04/16 13:05:35.278502 DEBUG:  Selected run image heroku/pack:20
 │ 2021/04/16 13:05:41.728675 DEBUG:  Pulling image heroku/pack:20
 │ 20: Pulling from heroku/pack
 │ Digest: sha256:ba0617c7610147b4997d1fd977d1b53f2054246e9c32bbc2af45a84e7eb064f4
 │ Status: Image is up to date for heroku/pack:20
 │ 2021/04/16 13:05:42.809341 DEBUG:  Refreshing registry cache for github.com//bui
 │ ldpacks/registry-index
 │ 2021/04/16 13:05:42.809376 DEBUG:  Validating registry cache for github.com//bui
 │ ldpacks/registry-index
 │ 2021/04/16 13:05:42.828276 DEBUG:  Creating registry cache for github.com//build
 │ packs/registry-index
! locating in registry urn:cnb:registry:fagiani/apt: reading entry: could
  not find buildpack: fagiani/apt: stat
/Users/nateberkopec/.pack/registry-a932275bd19c2d9e1b88fa06698fd2f5427a363d25bf87fa50
  no such file or directory
```

The buildpack in question is [valid and the URI is correct.](https://registry.buildpacks.io/buildpacks/fagiani/apt)

I can even see it in my local CNB registry:

```
❯ cat /Users/nateberkopec/.pack/registry-a932275bd19c2d9e1b88fa06698fd2f5427a363d25bf87fa500691c373089381/3/ap/fagiani_apt 
{"ns":"fagiani","name":"apt","version":"0.1.0","yanked":false,"addr":"ghcr.io/fagiani/buildpacks/fagiani_apt@sha256:622d013de847800ee7030cb839f459146e07771f8dacdf89318c25ee749b73b6"}
```

**Comment by nateberkopec** [comment]
I have no idea why but this may be specific to the buildpack in question - I can refer to, e.g. `urn:cnb:registry:heroku/ruby` and it works fine.

Hard to tell from the error message though if this is the buildpacks fault or Waypoint's.

**Comment by briancain** [comment]
@nateberkopec - The error looks to be coming from buildpack specifically. I'm not sure if it's just an issue with the pack builder client function or what, but it's interesting that the error doesn't have the full folder for where the pack should be living:

```bash
# the error
/Users/nateberkopec/.pack/registry-a932275bd19c2d9e1b88fa06698fd2f5427a363d25bf87fa50
```

Versus

```bash
# the actual dir
/Users/nateberkopec/.pack/registry-a932275bd19c2d9e1b88fa06698fd2f5427a363d25bf87fa500691c373089381
```

I'll go ahead and mark this as a bug for now and we can try to reproduce. Thanks for reporting the issue!

**Comment by nateberkopec** [comment]
Yeah, I couldn't figure out if that was just an artifact of how waypoint prints to the terminal or not.

**Comment by briancain** [comment]
Ah, that's a good point @nateberkopec - We've had a bug reported about our terminal printer cutting off long strings, so that might be why!

**Comment by fagiani** [comment]
> @nateberkopec - The error looks to be coming from buildpack specifically. I'm not sure if it's just an issue with the pack builder client function or what, but it's interesting that the error doesn't have the full folder for where the pack should be living:

@briancain -  I've been able to run the buildpack directly with `pack build` CLI as described [here](https://github.com/fagiani/apt-buildpack/issues/1#issuecomment-825094443)

Being the buildpack's maintainer, I'd be willing to help fixing or improving in a way it works with `Waypoint` if you could provide any general guidance or sending a PR.

Please feel free to reach out.

Keep Rocking!

**Comment by briancain** [comment]
Hey @fagiani ! Thanks for the extra information :) In that case we'll do some verification on our end to reproduce and see what is going wrong in Waypoint. Appreciate the help ❤️

**Comment by evanphx** [comment]
Hi @fagiani and @nateberkopec! I'm tracking this bug down and I've got a few questions:

1. @fagiani in the case where you have it working with the `pack` CLI tool, do you also have a project toml file? The pack CLI appears to only accept buildpacks that are configured in the project's toml file (which also provides the ability to indicate more information about each buildpack).
2. @nateberkopec Did you try using the `pack` CLI tool? If so, did it work?

**Comment by nateberkopec** [comment]
@evanphx we did get it working with `pack`, see https://github.com/fagiani/apt-buildpack/issues/1#issuecomment-827948526
